tensorflow/java/src/main/java/org/tensorflow/Tensor.java

```diff
@@ -34,6 +41,352 @@
  * }</pre>
  */
 public final class Tensor implements AutoCloseable {
+
+  /**
+   * Create a Tensor with data from the given buffer.
+   *
+   * <p>Supports the following datatypes: {@link DataType#INT32}
+   *
+   * <p>This method copies
+   * <i>n</i>&nbsp;=&nbsp;<tt>src.remaining()</tt> elements from the given
+   * buffer into the tensor, starting at the buffer's current position.
+   * The position of the buffer is then incremented by <i>n</i>.
+   *
+   * @param dataType the tensor datatype.
+   * @param shape the tensor shape.
+   * @param data a buffer containing the tensor data.
+   *
+   * @throws  IllegalArgumentException
+   *          If the tensor datatype or shape is not compatible with the buffer
+   */
+  public static Tensor create(DataType dataType, long[] shape, IntBuffer data) {
+    if(dataType != DataType.INT32) {
+      throw incompatibleBuffer(dataType);
+    }
+    if(data.remaining() != numElements(shape)) {
+      throw incompatibleBuffer(shape);
+    }
+    int elemSize = Integer.SIZE / Byte.SIZE;
```

Comment:
Was unable to do this, tried to but the `ByteBuffer` variant was a special case.